pkg/sbom/collectors/containerd/containerd.go

```diff
@@ -40,7 +41,7 @@ func (r ScanRequest) Collector() string {
 
 // Type returns the scan request type
 func (r ScanRequest) Type() string {
-	if config.Datadog.GetBool("sbom.container_image.use_mount") {
+	if pkgconfig.Datadog.GetBool("sbom.container_image.use_mount") {
```

Comment:
Do you think we can eliminate this call here? 

This is the only line in this file where `pkgconfig` is used. 

How about the following:
- Storing the config component as a dependency in `Scanner`, and passing the config down to `Type` method of `ScanRequest`
- Adding a `UseMount` property to the scanner struct, which is initialised to `sbom.container_image.use_mount` using the config component, and then we can pass that property directly to `Type`

My intention is to fully eliminate the usage of `pkg/config` here.

WDYT?

Reply:
I think it should work, nice catch 👍